web/react/utils/text_formatting.jsx

```diff
@@ -123,27 +124,60 @@ function autolinkUrls(text, tokens) {
 }
 
 function autolinkAtMentions(text, tokens) {
-    let output = text;
+    // Return true if provided character is punctuation
+    function isPunctuation(character) {
+        const re = /[\u2000-\u206F\u2E00-\u2E7F\\'!"#$%&()*+,\-.\/:;<=>?@\[\]^_`{|}~]/g;
```

Comment:
I'm not a regex guru but I believe dots in [ ] brackets do not need to be escaped. Check out https://jsfiddle.net/21b1su24/